resip/recon/UserAgentClientPublication.cxx

```diff
@@ -0,0 +1,113 @@
+#include "UserAgent.hxx"
+#include "ReconSubsystem.hxx"
+#include "UserAgentClientPublication.hxx"
+
+#include <rutil/Log.hxx>
+#include <rutil/Logger.hxx>
+#include <resip/dum/DialogUsageManager.hxx>
+#include <resip/dum/ClientPublication.hxx>
+
+using namespace recon;
+using namespace resip;
+using namespace std;
+
+#define RESIPROCATE_SUBSYSTEM ReconSubsystem::RECON
+
+UserAgentClientPublication::UserAgentClientPublication(UserAgent& userAgent, DialogUsageManager& dum, unsigned int handle)
+: AppDialogSet(dum),
+  mUserAgent(userAgent),
+  mDum(dum),
+  mPublicationHandle(handle),
+  mEnded(false)
+{
+   mUserAgent.registerPublication(this);
+}
+
+UserAgentClientPublication::~UserAgentClientPublication()
+{
+   mUserAgent.unregisterPublication(this);
+}
+
+PublicationHandle 
+UserAgentClientPublication::getPublicationHandle()
+{
+   return mPublicationHandle;
+}
+
+void 
+UserAgentClientPublication::end()
+{
+   if(!mEnded)
+   {
+      mEnded = true;
+      AppDialogSet::end();
+   }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// ClientPublicationHandler ///////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+void
+UserAgentClientPublication::onSuccess(ClientPublicationHandle handle, const SipMessage& status)
+{
+   InfoLog(<<"UserAgentClientPublication::onSuccess - not implemented\n");
+}
+
+void
+UserAgentClientPublication::onRemove(ClientPublicationHandle handle, const SipMessage& status)
+{
+   InfoLog(<<"UserAgentClientPublication::onRemove - not implemented\n");
+}
+
+void
+UserAgentClientPublication::onFailure(ClientPublicationHandle handle, const SipMessage& status)
+{
+   InfoLog(<<"UserAgentClientPublication::onFailure - not implemented\n");
+}
+
+int
+UserAgentClientPublication::onRequestRetry(ClientPublicationHandle handle, int retrySeconds, const SipMessage& status)
+{
+   InfoLog(<<"UserAgentClientPublication::onRequestRetry - not implemented\n");
+   return 30;
+}
+
+void
+UserAgentClientPublication::onStaleUpdate(ClientPublicationHandle handle, const SipMessage& status)
+{
+   InfoLog(<<"UserAgentClientPublication::onStaleUpdate - not implemented\n");
+}
+
+/* ====================================================================
+
+ Copyright (c) 2007-2008, Plantronics, Inc.
+ All rights reserved.
+
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions are 
+ met:
+
+ 1. Redistributions of source code must retain the above copyright 
+    notice, this list of conditions and the following disclaimer. 
+
+ 2. Redistributions in binary form must reproduce the above copyright
+    notice, this list of conditions and the following disclaimer in the
+    documentation and/or other materials provided with the distribution. 
+
+ 3. Neither the name of Plantronics nor the names of its contributors 
```

Comment:
instead of "Plantronics", it should say "the authors", then it is easier to cut and paste too.  You can copy the whole license block from resip/recon/SipXHelper.cxx and use it in any new files you add to the project